easybuild/easyconfigs/d/dictys/dictys-1.1.0-foss-2023a-CUDA-12.1.1.eb

```diff
@@ -0,0 +1,102 @@
+easyblock = 'PythonBundle'
+
+name = 'dictys'
+version = '1.1.0'
+versionsuffix = '-CUDA-%(cudaver)s'
+
+homepage = 'https://github.com/pinellolab/dictys'
+description = "Context specific and dynamic gene regulatory network reconstruction and analysis."
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+
+builddependencies = [('poetry', '1.5.1')]
+dependencies = [
+    ('Python', '3.11.3'),
+    ('CUDA', '12.1.1', '', SYSTEM),
+    ('PyTorch-bundle', '2.1.2', versionsuffix),
+    ('pybedtools', '0.9.1'),
+    ('SAMtools', '1.18'),
+    ('MACS2', '2.2.9.1'),
+    ('FFmpeg', '6.0'),
+    ('matplotlib', '3.7.2'),
+    ('Python-bundle-PyPI', '2023.06'),
+    ('SciPy-bundle', '2023.07'),
+    ('networkx', '3.1'),
+    ('h5py', '3.9.0'),
+    ('pyro-ppl', '1.9.0', versionsuffix),
+    ('adjustText', '0.7.3'),
+    ('Pysam', '0.22.0'),
+    ('paramiko', '3.2.0'),
+    ('Jupyter-bundle', '20230823'),
+    ('Qtconsole', '5.5.1'),
+]
+
+# regenerate WellingtonC.c to works with python 3.11 + unpin matplotlib version
+local_pyDNase_preinstallopts = (
+    "cd pyDNase/footprinting && rm WellingtonC.c && cythonize -i WellingtonC.pyx && cd .. && cd .. && "
+    "sed -i 's/matplotlib < 2.0.0/matplotlib/' setup.py && "
+)
+
+exts_list = [
+    ('jupyter_console', '6.6.3', {
+        'source_tmpl': SOURCE_PY3_WHL,
+        'checksums': ['309d33409fcc92ffdad25f0bcdf9a4a9daa61b6f341177570fdac03de5352485'],
+    }),
+    ('jupyter', '1.0.0', {
+        'checksums': ['d9dc4b3318f310e34c82951ea5d6683f67bed7def4b259fafbfe4f1beb1d8e5f'],
+    }),
+    ('args', '0.1.0', {
+        'checksums': ['a785b8d837625e9b61c39108532d95b85274acd679693b71ebb5156848fcf814'],
+    }),
+    ('clint', '0.5.1', {
+        'checksums': ['05224c32b1075563d0b16d0015faaf9da43aa214e4a2140e51f08789e7a4c5aa'],
+    }),
+    ('pynetbox', '7.4.0', {
+        'checksums': ['fd0b1f197b3880048408ff5ed84422dd599bcd9389e32cb06a09b9b0d55c1636'],
+    }),
+    ('yamlordereddictloader', '0.4.2', {
+        'checksums': ['36af2f6210fcff5da4fc4c12e1d815f973dceb41044e795e1f06115d634bca13'],
+    }),
+    ('transitions', '0.9.2', {
+        'checksums': ['2f8490dbdbd419366cef1516032ab06d07ccb5839ef54905e842a472692d4204'],
+    }),
+    ('scp', '0.15.0', {
+        'checksums': ['f1b22e9932123ccf17eebf19e0953c6e9148f589f93d91b872941a696305c83f'],
+    }),
+    ('pyserial', '3.5', {
+        'modulename': 'serial',
+        'checksums': ['3c77e014170dfffbd816e6ffc205e9842efb10be9f58ec16d3e8675b4925cddb'],
+    }),
+    ('ncclient', '0.6.15', {
+        'checksums': ['6757cb41bc9160dfe47f22f5de8cf2f1adf22f27463fb50453cc415ab96773d8'],
+    }),
+    ('junos-eznc', '2.7.0', {
```

Comment:
@pavelToman As discussed, let's add a standalone easyconfig for `junos-eznc`, and use the latest version (2.7.1) in which we install their `paramiko` fork as an extra extension, and remove the `pip install` command from their `setup.py`.

I've also opened an issue upstream on this, see https://github.com/Juniper/py-junos-eznc/issues/1318